pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/impl/updateprofile/UpdateProfileProcessor.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.ddp.event.dsmtask.impl.updateprofile;
+
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskData;
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskProcessorAbstract;
+
+/**
+ * Processes Dsm Task of taskType='UPDATE_PROFILE'.
+ * It updates person's email, first, and last name.
+ *
+ * <p>Does the following steps:
+ * <ul>
+ *     <li>updates user's email (checking if this email not used by another user);</li>
+ *     <li>updates user's first and last name.</li>
+ * </ul>
+ * NOTE: if any data (passed from Dsm) is null then it not updated.
+ */
+public class UpdateProfileProcessor extends DsmTaskProcessorAbstract {
+
+    public static final String TASK_TYPE__UPDATE_PROFILE = "UPDATE_PROFILE";
```

Comment:
I think task type should be a `TaskType` enum instead of strings.

Reply:
I did it for purpose (not using enum): in order to separate api and impl layer: in case of adding type in enum it needs to create this enum in api layer. Currently it is possible no to change api layer when need to add new type/new processor.
I posted more detailed message in Slack..
If you think that it not worth do do efforts for components loose coupling, then I can define types as enums. 
It's not a problem: I just want to understand architect approach in Pepper project and will follow it (I'm not a 'religious' regarding coding styles:)).

Reply:
Okay, let's give that a try!